packages/marko-web-theme-monorail/browser/inline-newsletter-form.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="inline-newsletter-form">
+    <div ref="lazyload" class="lazyload" />
+    <step-1
+      v-if="step === 1"
+      :newsletter="defaultNewsletter"
+      :name="name"
+      :description="description"
+      :disabled="disabled"
+      :image-src="imageSrc"
+      :image-srcset="imageSrcset"
+      :recaptcha-site-key="recaptchaSiteKey"
+      :privacy-policy-link="privacyPolicyLink"
+      :lang="lang"
+      @submit="stepOneSubmit"
+      @subscribe="$emit('subscribe', $event)"
+      @focus="$emit('focus', { step: 1 })"
+      @error="$emit('error', { step: 1, error: $event })"
+    />
+    <step-2
+      v-if="step === 2"
+      :site-name="siteName"
+      :email="email"
+      :default-newsletter="defaultNewsletter"
+      :newsletters="newsletters"
+      :demographic="demographic"
+      :recaptcha-site-key="recaptchaSiteKey"
+      :lang="lang"
+      as-card
+      @submit="$emit('submit', { step: 2 })"
+      @subscribe="$emit('subscribe', $event)"
+      @focus="$emit('focus', { step: 2 })"
+      @load="$emit('load', { step: 2 })"
+      @error="$emit('error', { step: 2, error: $event })"
+    />
+  </div>
+</template>
+
+<script>
+import Step1 from './inline-newsletter-form/step1.vue';
+import Step2 from './newsletter-signup-form/step2.vue';
+
+export default {
+  components: {
+    Step1,
+    Step2,
+  },
+
+  props: {
+    siteName: {
+      type: String,
+      required: true,
+    },
+    recaptchaSiteKey: {
+      type: String,
+      required: true,
+    },
+    name: {
+      type: String,
+      required: true,
+    },
+    description: {
+      type: String,
+      required: true,
+    },
+    defaultNewsletter: {
+      type: Object,
+      required: true,
+      validate: o => (o && o.name && o.deploymentTypeId),
+    },
+    newsletters: {
+      type: Array,
+      default: () => [],
+    },
+    demographic: {
+      type: Object,
+      required: true,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    imageSrc: {
+      type: String,
+      default: null,
+    },
+    imageSrcset: {
+      type: String,
+      default: null,
+    },
+    privacyPolicyLink: {
+      type: Object,
+      required: true,
+    },
+    lang: {
+      type: String,
+      default: 'en',
+    },
+  },
+
+  data: () => ({
+    didView: false,
+    email: null,
+    step: 1,
+  }),
+
+  watch: {
+    didView(value) {
+      if (value) this.$emit('view', { step: 1 });
+    },
+  },
+
+  mounted() {
+    this.$emit('load', { step: 1 });
+    this.$refs.lazyload.addEventListener('lazybeforeunveil', () => {
+      this.didView = true;
+    });
+  },
+
+  methods: {
+    stepOneSubmit({ email, encryptedCustomerId }) {
+      this.$emit('submit', { step: 1 });
+      this.email = email;
+      this.step = 2;
+      if (window.olytics) window.olytics.confirm(encryptedCustomerId);
```

Comment:
Don't run this inside the component; add this logic to your `submit` event binding in the browser.js file (around L97). This is done in another component as well, move it from there too

Reply:
@solocommand I moved line 125 into the index.js login with the following changes. and made the same change to site-newsletter-menu.vue as well.  

```
Browser.register('ThemeInlineNewsletterForm', InlineNewsletterForm, {
    on: {
      load: data => emitNewsletterEvent({ type: 'Inline', action: 'Load', data }),
      view: data => emitNewsletterEvent({ type: 'Inline', action: 'View', data }),
      focus: data => emitNewsletterEvent({ type: 'Inline', action: 'Focus', data }),
      ###
      submit: (data) => {
        emitNewsletterEvent({ type: 'Inline', action: 'Submit', data });
        if (window.olytics) window.olytics.confirm(data.encryptedCustomerId);
      },
      ###
      subscribe: ({ newsletter }) => emitNewsletterSubscription({ type: 'Pushdown', newsletter }),
      error: data => emitNewsletterEvent({ type: 'Inline', action: 'Error', data: { ...data, error: data.error.message } }),
    },
  });
  ```